engine/util/src/main/scala/pl/touk/nussknacker/engine/util/cache/DefaultCache.scala

```diff
@@ -11,69 +10,37 @@ import scala.concurrent.duration.Duration
   * @param expireAfterAccess the expiration time from last action (read / write)
   * @tparam T
   */
-class DefaultCache[T](maximumSize: Long, expireAfterAccess: Option[Duration], registerStats: Boolean) extends Cache[T] {
+class DefaultCache[T](maximumSize: Long, expireAfterAccess: Option[Duration]) extends Cache[T] {
 
-  import scalacache.caffeine._
-  import scalacache.modes.sync._
+  import scala.compat.java8.FunctionConverters._
 
-  import scala.concurrent.duration._
-
-  private val stats = new ConcurrentStatsCounter
-
-  private lazy val caffeineClientCache = {
+  private lazy val underlying: cache.Cache[String, T] = {
     val builder = Caffeine
       .newBuilder()
+      .asInstanceOf[Caffeine[String, T]]
+
+    builder
       .maximumSize(maximumSize)
       .weakValues()
 
-    expireAfterAccess.foreach(expire => builder.expireAfterAccess(java.time.Duration.ofMillis(expire.toMillis)))
-
-    builder.build[String, Entry[T]]
-  }
-
-  implicit private val cache: CaffeineCache[T] =
-    CaffeineCache(caffeineClientCache)
+    expireAfterAccess
+      .foreach(expire => builder.expireAfterAccess(java.time.Duration.ofMillis(expire.toMillis)))
```

Comment:
We should use `expireAfterWrite`, because you can has many read accesses but still want to invalidate after a some delay from write...

Reply:
I will add these three options..

Reply:
You are right but.. Now I see thaht we should set expireAfterWrite for latest and expireAfterAccess for versions. I've added it.